Source/common/SNTFileInfo.m

```diff
@@ -642,7 +642,9 @@ - (struct mach_header *)firstMachHeader {
 ///
 - (NSData *)safeSubdataWithRange:(NSRange)range {
   @try {
-    if ((range.location + range.length) > self.fileSize) return nil;
+    if (range.location > self.fileSize || range.length > self.fileSize ||
+        (range.location + range.length) > self.fileSize)
+      return nil;
```

Comment:
Is the goal here to make this safer for large values? Seems there could still be an issue on the upper boundaries - this issue previously existed as well... Maybe should change to something like:

```suggestion
    NSUInteger size;
    if (__builtin_add_overflow(range.location, range.length, &size) || size > self.fileSize) {
      return nil;
    }
```

Reply:
Yeah this arose from size being near UINT64_MAX. Changed.